src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,323 @@
+import org.prebid.server.analytics.AnalyticsReporter;
+import org.prebid.server.analytics.model.AmpEvent;
+import org.prebid.server.analytics.model.AuctionEvent;
+import org.prebid.server.analytics.reporter.greenbids.model.AdUnit;
+import org.prebid.server.analytics.reporter.greenbids.model.CommonMessage;
+import org.prebid.server.analytics.reporter.greenbids.model.ExtBanner;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsAnalyticsProperties;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsBidder;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsConfig;
+import org.prebid.server.analytics.reporter.greenbids.model.HttpUtil;
+import org.prebid.server.analytics.reporter.greenbids.model.JsonUtil;
+import org.prebid.server.analytics.reporter.greenbids.model.MediaTypes;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.auction.model.BidRejectionReason;
+import org.prebid.server.auction.model.BidRejectionTracker;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtStoredRequest;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.NonBid;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.SeatNonBid;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.UUID;
+
+public class GreenbidsAnalyticsReporter implements AnalyticsReporter {
+
+    private static final String ANALYTICS_SERVER = "https://a.greenbids.ai/";
+    public String pbuid;
+    public Double greenbidsSampling;
+    public Double exploratorySamplingSplit;
+    private final GreenbidsConfig greenbidsConfig;
+    private final JacksonMapper jacksonMapper;
+    public Boolean isSampled;
+    public String greenbidsId;
+    public String billingId;
+
+    public GreenbidsAnalyticsReporter(
+            GreenbidsAnalyticsProperties greenbidsAnalyticsProperties,
+            JacksonMapper jacksonMapper
+    ) {
+        this.pbuid = Objects.requireNonNull(greenbidsAnalyticsProperties.getPbuid());
+        this.greenbidsSampling = greenbidsAnalyticsProperties.getGreenbidsSampling();
+        this.exploratorySamplingSplit = 0.9;
+        this.greenbidsConfig = GreenbidsConfig.of(
+                greenbidsAnalyticsProperties.getPbuid(),
+                greenbidsAnalyticsProperties.getGreenbidsSampling()
+        );
+        this.jacksonMapper = Objects.requireNonNull(jacksonMapper);
+        this.isSampled = null;
+        this.greenbidsId = null;
+        this.billingId = null;
```

Comment:
please make them local variable in the place they're used

Reply:
fixed, they are used in `processEvent` and in builder of `CommonMessage`

Reply:
fixed